### PR TITLE
[Feat7-FR2]: Payment Schema + Test + Documentation

### PR DESCRIPTION
**Pull request by fabihafifa**
Implements payment schema models using Pydantic, including PaymentSchema, UpdatePaymentSchema, PaymentStatus, and CostBreakdown. Adds validation for required fields, input constraints, and cost consistency, along with unit tests for initialization, validation errors, and partial updates.

Methods: (payment_schema.py) PaymentStatus, Cost breakdown, PaymentBase, PaymentCreate, Payment, PaymentUpdate, all schema validators in this file.

Tests: (test_payment_schema.py) schema/model validation tests only.

Disclaimer: complexity will be added later. For now PR addresses basic requirements.

## Changes
- [x] New feature to new file
- [x] Documentation

## Checklist
- [x] All tests pass
- [x] Test documentation submitted to folder
- [x] Added necessary comments
- [x] Formatted correctly

### CODE REVIEW

File: backend/models/payment/payment_schema.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    amount: Optional[float] = Field(default=None, alias="_amount")
+
+    @field_validator("card_name")
+    @classmethod
+    def validate_card_name(cls, value: Optional[str]) -> Optional[str]:
+        if value is not None and not value.strip():
+            raise ValueError("card_name cannot be blank")
+        return value.strip() if isinstance(value, str) else value
+
+    @field_validator("card_number")
```
Comment (bradyb2005) [inline backend/models/payment/payment_schema.py]: add to this validator: card numbers must be exclusively numbers, and can range from a length of 13 characters to 19 characters. To check this as an int, the best way is to wrap it in a str to check the length. Kind of like this:

len(str(my_num)) >= 13

File: backend/models/payment/payment_schema.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    @field_validator("security_number")
+    @classmethod
+    def validate_security_number(cls, value: Optional[int]) -> Optional[int]:
+        if value is not None and value <= 0:
+            raise ValueError("security_number must be positive")
+        return value
+
+    @field_validator("expiration")
+    @classmethod
+    def validate_expiration(cls, value: Optional[str]) -> Optional[str]:
```
Comment (bradyb2005) [inline backend/models/payment/payment_schema.py]: Add this to the validator: check validity of the expiration date string by using regex. Here's an if statement that would do this:

if not re.match(r"^(0[1-9]|1[0-2])\/\d{2}$", value.strip()): 

Here's how that regex works:
`(0[1-9]|1[0-2])`: The | is showing that there's two options for the first 2 characters. the first option is if the first character is 0, the second character has to be between 1-9. second option is if the first character is 1, the second character must be between 0-2.
`\/\`: Two escape characters to say that the "/" character must be next.
`d{2}`: the next 2 characters have to be numbers.
`$`: this just means that has to be the end of the string, enforcing the length.

After that passes, you can also take out the month and year into their own integers and append 2000 to the collected year. By using datetime.now() you can check if the expiration has passed. Doing all of this will successfully check that expiration is valid!

File: backend/tests/payment/unit_tests/test_payment_schema.py
Comment (bradyb2005) [inline backend/tests/payment/unit_tests/test_payment_schema.py]: Here are some additional things that should be tested when implementing the changes I requested in payment_schema. You can use `@pytest.mark.parameterize` to have a test go through multiple cases seperately without repeating code:

**Card Number:**
Edge Cases: Test where length is 13 and 19 (pass)
Outside of boundary cases: Test where length is less than 13 and greater than 19 (fail)

**Expiration Date:**
Because of how regex works, the proper way to test it is to go through each check in order. Here's how:
(note, the ONLY part of each test that should be violated is the specific case. For example, a good example for Case 1 would be 21/12. All other checks in the regex would pass except case 1)
Case 1 - 1st character is not 0 or 1 (fail)
Case 2 - 1st character is 0, next character is not 1-9 (fail)
Case 3 - 1st character is 1, next character is not 0-2 (fail)
Case 4 - characters 1-2 are either of the valid options, 3rd character is not / (fail)
Case 5 - characters 1-3 are valid, 4th character is not a number (this case doesn't need to be repeated for 5th character because the regex statement is for the next 2 characters) (fail)
Case 6 - regex tests pass, month and year are exactly today's month and year (edge case, should fail)
Case 7 - regex tests pass, month is a previous month, and year is current year (fail)
Case 8 - regex tests pass, month is current month or greater, year is past year (should fail)
Case 9 - regex tests pass and expiration date is in the future (pass)

**Security Code:**
Edge Cases: Test where length is 3 and 4 (pass)
Outside of Boundary cases: Test where lengths is less than 3 and greater than 4 (fail)

File: backend/models/payment/payment_schema.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+
+class PaymentBase(BaseModel):
+    model_config = ConfigDict(
+        arbitrary_types_allowed=True,
+        populate_by_name=True,
+        validate_assignment=True,
+    )
+
+    order: Optional[Any] = Field(default=None, alias="_order")
```
[the remark this answers is not in the document]
Comment (Graysonzz) [reply backend/models/payment/payment_schema.py]: I agree, this would be beneficial

File: backend/models/payment/payment_schema.py (hunk 1 above)
Comment (Graysonzz) [reply backend/models/payment/payment_schema.py]: I agree with this suggestion, how you do it is up to you

File: backend/models/payment/payment_schema.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    @field_validator("amount")
+    @classmethod
+    def validate_amount(cls, value: Optional[float]) -> Optional[float]:
+        if value is not None and value < 0:
+            raise ValueError("amount cannot be negative")
+        return float(value) if value is not None else value
+
+
+class PaymentSchema(PaymentBase):
```
Comment (Graysonzz) [inline backend/models/payment/payment_schema.py]: Remove schema

File: backend/tests/documentation/payment_schema/payment_schema.md
Comment (Graysonzz) [inline backend/tests/documentation/payment_schema/payment_schema.md]: If there's any way you can attach your picture to this file, that would be ideal to reduce clutter
